__tests__/app.test.js

```diff
@@ -183,3 +194,47 @@ describe("GET /api/articles", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("POST /api/articles/:article_id/comments responds with status code 201 and add a new comment in in the comments table with the given username and body", () => {
+    const newComment = { username: "rogersop", body: "what a lovely story!!" };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            body: expect.any(String),
+            votes: expect.any(Number),
+            author: expect.any(String),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+
+  test("POST /api/articles/:article_id/comments responds with status code 404 username is not a valid username in users table", () => {
+    const newComment = { username: "Hello", body: "what a lovely story!!" };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid username");
+      });
+  });
+
+  test("POST /api/articles/:article_id/comments responds with status code 404 article is not a valid article id from articles table", () => {
+    const newComment = { username: "rogersop", body: "what a lovely story!!" };
+    return request(app)
+      .post("/api/articles/222/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid article_id");
+      });
+  });
+});
```

Comment:
Could we have a test for;
- if the article_id is not a number?
- if the body is missing? 
- if the username is missing?